File: mekatrol.automatum.editor/src/flow/types/FlowConnection.ts
export interface FlowConnection {
  // ID of the block the connection starts from
  startBlockId: string;

  // The pin number of the input/output that this connection starts from
  startPin: number;

  // ID of the block the connection ends at
  endBlockId: string;

  // The pin number of the input/output that this connection ends at
  endPin: number;

  // True if currently selected
  selected: boolean;
}

File: Mekatrol.Automatum/Mekatrol.Automatum.Models/Flows/FlowConnection.cs
namespace Mekatrol.Automatum.Models.Flows;

public class FlowConnection : BaseModel
{
    // Connection configiration properties
    public Guid StartBlockId { get; set; } = Guid.Empty;

    public int StartPin { get; set; }

    public Guid EndBlockId { get; set; } = Guid.Empty;

    public int EndPin { get; set; }

    // Visual properties for display in user interface
    public bool Selected { get; set; }
}